File: python_sdk/src/reality_capture/specifications/touchup.py
from pydantic import BaseModel, Field
from typing import Optional
from enum import Enum


class TouchUpExportInputs(BaseModel):
    modeling_reference: str = Field(alias="modelingReference", description="Reality data id of Modeling Reference")
    tiles_to_touch_up: Optional[list[str]] = Field(None, alias="tilesToTouchUp",
                                                   description="List of tiles to export for touch up")


class TouchLevel(Enum):
    GEOMETRY = "Geometry"
    GEOMETRY_AND_TEXTURE = "GeometryAndTexture"


class TouchUpExportOptions(BaseModel):
    level: Optional[TouchLevel] = Field(None, description="Touch Up level")
    crs: Optional[str] = Field(None, description="Export CRS")


class TouchUpExportOutputsCreate(Enum):
    TOUCH_UP_DATA = "touchUpData"


class TouchUpExportOutputs(BaseModel):
    touch_up_data: str = Field(alias="touchUpData", description="Reality Data id for touch up data")


class TouchUpExportSpecifications(BaseModel):
    inputs: TouchUpExportInputs = Field(description="Inputs")
    outputs: TouchUpExportOutputs = Field(description="Outputs")
    options: Optional[TouchUpExportOptions] = Field(None, description="Options")


class TouchUpExportSpecificationsCreate(BaseModel):
    inputs: TouchUpExportInputs = Field(description="Inputs")
    outputs: list[TouchUpExportOutputsCreate] = Field(description="Outputs")
    options: Optional[TouchUpExportOptions] = Field(None, description="Options")


class TouchUpExportCost(BaseModel):
    tile_count: int = Field(description="Number of tiles to export", alias="tileCount", ge=0)


class TouchUpImportInputs(BaseModel):
    modeling_reference: str = Field(alias="modelingReference", description="Reality data id of Modeling Reference")
    touch_up_data: str = Field(alias="touchUpData", description="Reality Data id for touch up data")


class TouchUpImportOutputsCreate(Enum):
    IMPORT_INFO = "importInfo"


class TouchUpImportOutputs(BaseModel):
    import_info: Optional[str] = Field(None, alias="importInfo", description="Folder in bucket containing the "
                                                                             "information about what was imported",
                                       pattern=r"^bkt:.+")


class TouchUpImportSpecificationsCreate(BaseModel):
    inputs: TouchUpImportInputs = Field(description="Inputs")
    outputs: list[TouchUpImportOutputsCreate] = Field(description="Outputs")


class TouchUpImportSpecifications(BaseModel):
    inputs: TouchUpImportInputs = Field(description="Inputs")
    outputs: TouchUpImportOutputs = Field(description="Outputs")


class TouchUpImportCost(BaseModel):
    tile_count: int = Field(description="Number of tiles to import", alias="tileCount", ge=0)


class ImportTileInfo(BaseModel):
    tile_name: str = Field(alias="tileName", description="Name of an imported tile")
    level: TouchLevel = Field(description="Touch up level of imported tile")


class ImportInfo(BaseModel):
    import_info: list[ImportTileInfo] = Field(alias="importInfo", description="List of tiles imported")

File: typescript_sdk/reality_capture/src/specifications/touchup.ts
import { z } from "zod";

export enum TouchLevel {
  GEOMETRY = "Geometry",
  GEOMETRY_AND_TEXTURE = "GeometryAndTexture"
}

export enum TouchUpExportOutputsCreate {
  TOUCH_UP_DATA = "touchUpData"
}

export enum TouchUpImportOutputsCreate {
  IMPORT_INFO = "importInfo"
}

export const TouchUpExportInputsSchema = z.object({
  modelingReference: z.string().describe("Reality data id of Modeling Reference"),
  tilesToTouchUp: z.array(z.string()).optional().describe("List of tiles to export for touch up"),
});
export type TouchUpExportInputs = z.infer<typeof TouchUpExportInputsSchema>;

export const TouchUpExportOptionsSchema = z.object({
  level: z.nativeEnum(TouchLevel).optional().describe("Touch Up level"),
  crs: z.string().optional().describe("Export CRS"),
});
export type TouchUpExportOptions = z.infer<typeof TouchUpExportOptionsSchema>;

export const TouchUpExportOutputsSchema = z.object({
  touchUpData: z.string().describe("Reality Data id for touch up data"),
});
export type TouchUpExportOutputs = z.infer<typeof TouchUpExportOutputsSchema>;

export const TouchUpExportSpecificationsSchema = z.object({
  inputs: TouchUpExportInputsSchema.describe("Inputs"),
  outputs: TouchUpExportOutputsSchema.describe("Outputs"),
  options: TouchUpExportOptionsSchema.optional().describe("Options"),
});
export type TouchUpExportSpecifications = z.infer<typeof TouchUpExportSpecificationsSchema>;

export const TouchUpExportSpecificationsCreateSchema = z.object({
  inputs: TouchUpExportInputsSchema.describe("Inputs"),
  outputs: z.array(z.nativeEnum(TouchUpExportOutputsCreate)).describe("Outputs"),
  options: TouchUpExportOptionsSchema.optional().describe("Options"),
});
export type TouchUpExportSpecificationsCreate = z.infer<typeof TouchUpExportSpecificationsCreateSchema>;

export const TouchUpExportCostSchema = z.object({
  tileCount: z.number().min(0).describe("Number of tiles to export"),
});
export type TouchUpExportCost = z.infer<typeof TouchUpExportCostSchema>;

export const TouchUpImportInputsSchema = z.object({
  modelingReference: z.string().describe("Reality data id of Modeling Reference"),
  touchUpData: z.string().describe("Reality Data id for touch up data"),
});
export type TouchUpImportInputs = z.infer<typeof TouchUpImportInputsSchema>;

export const TouchUpImportOutputsSchema = z.object({
  importInfo: z
    .string()
    .regex(/^bkt:.+/, { message: "Must start with 'bkt:'" })
    .optional()
    .describe("Folder in bucket containing the information about what was imported"),
});
export type TouchUpImportOutputs = z.infer<typeof TouchUpImportOutputsSchema>;

export const TouchUpImportSpecificationsCreateSchema = z.object({
  inputs: TouchUpImportInputsSchema.describe("Inputs"),
  outputs: z.array(z.nativeEnum(TouchUpImportOutputsCreate)).describe("Outputs"),
});
export type TouchUpImportSpecificationsCreate = z.infer<typeof TouchUpImportSpecificationsCreateSchema>;

export const TouchUpImportSpecificationsSchema = z.object({
  inputs: TouchUpImportInputsSchema.describe("Inputs"),
  outputs: TouchUpImportOutputsSchema.describe("Outputs"),
});
export type TouchUpImportSpecifications = z.infer<typeof TouchUpImportSpecificationsSchema>;

export const TouchUpImportCostSchema = z.object({
  tileCount: z.number().min(0).describe("Number of tiles to import"),
});
export type TouchUpImportCost = z.infer<typeof TouchUpImportCostSchema>;

export const ImportTileInfoSchema = z.object({
  tileName: z.string().describe("Name of an imported tile"),
  level: z.nativeEnum(TouchLevel).describe("Touch up level of imported tile"),
});
export type ImportTileInfo = z.infer<typeof ImportTileInfoSchema>;

export const ImportInfoSchema = z.object({
  importInfo: z.array(ImportTileInfoSchema).describe("List of tiles imported"),
});
export type ImportInfo = z.infer<typeof ImportInfoSchema>;